src/features/workout-builder/model/workout-builder.store.ts

```diff
@@ -90,6 +91,17 @@ export const useWorkoutBuilderStore = create<WorkoutBuilderState>((set, get) =>
 
   setExercisesOrder: (order) => set({ exercisesOrder: order }),
 
+  deleteExercise: (exerciseId) =>
+    set((state) => ({
+      exercisesByMuscle: state.exercisesByMuscle
+        .map((group) => ({
+          ...group,
+          exercises: group.exercises.filter((ex: any) => ex.id !== exerciseId),
+        }))
+        .filter((group) => group.exercises.length > 0),
+      exercisesOrder: state.exercisesOrder.filter((id) => id !== exerciseId),
+    })),
+
```

Comment:
We could avoid recreating every group of muscles if the length is the same with something like this :

```tsx
  deleteExercise: (exerciseId) =>
    set((state) => ({
      exercisesByMuscle: state.exercisesByMuscle
        .map((group) => {
          const filteredExercises = group.exercises.filter((ex: any) => ex.id !== exerciseId);

         // with this check ↓
          if (filteredExercises.length === group.exercises.length) {
            return group;
          }

          return { ...group, exercises: filteredExercises };
        })
        .filter((group) => group.exercises.length > 0),
      exercisesOrder: state.exercisesOrder.filter((id) => id !== exerciseId),
    })),
```